### ISSUE TITLE
Can nopt not handle optional arguments to known options?

### ISSUE THREAD

**Issue by jasonkarns** [npm/nopt#65]
We have known options:

```
var knownOptions = {
    '508': String,
    'wcag2': String,
    'devmin': String
};
```

None of which _require_ an argument.

`cli --508 arg508 --wcag argWcag` works fine. It results in:
`{508: "arg508", wcag: "argWcag"}`

However, `cli --508 --wcag` results in:
`{508: "--wcag"}`

but we want:
`{508: "", wcag: ""}`

We've tried making these options be of a custom type that validates the value is not itself a known option. However, that doesn't push the option pack into the parsed options; it just invokes the invalidOption handler and exits.

Is there a way to get this behavior such that the argument to the known options is optional?

**Comment by jasonkarns** [comment]
I also attempted defining shorthands:

```
var knownOptions = {
	'508': String,
	'wcag2': String
};
var shortHands = {
	'508': ['--508', 'default508'],
	'wcag':['--wcag', 'defaultWCAG']
};
```
 Expecting the shorthands (when missing the `=`) to expand, but to be left alone with set explicitly with `=`.

Expected: `cli --508=1 --wcag=2` => `{508: "1", wcag: "2"}`
but got: `{508: "1", wcag: "defaultWCAG"}`

which makes no sense. Apparently the option was expanded as a shorthand (even though it was supplied with an `=`?). But the first option was left as-is???

Also, Expected: `cli --508 --wcag` => `{508: "default508", wcag: "defaultWCAG"}`
but got: `{508: "--wcag"}`

**Comment by samjonester** [comment]
In a similar vein, Providing just one flag with no value has unexpected behavior as well.

Expected: cli --508 => {508: ""}
but got: {508: "true"}

**Comment by jasonkarns** [comment]
Sweet, thanks @samjonester !!
